### DISCUSSION TITLE
If I enable lfs, when folks pull the repository, do they still download the full file

### DISCUSSION THREAD

**Discussion by flexwang2** [git-lfs/git-lfs#5129] [Q&A]
We have some big files in our repository. If I track them using lfs, when someone else clone the repository. Do they just download the small pointer file or they still download the large files as well?

**Comment by bk2204** [comment]
Hey,

In general, if they have Git LFS configured in the default way, they clone the repository with Git, which downloads the pointer files, and then during checkout Git LFS is invoked and only the large files needed for the checkout are pulled and checked out into the working tree.  So the user ends up with the large files needed for the checkout, but only those large files.

**Reply by flexwang2** [reply]
Thanks for answering. However, when I set up the lfs in my git and then `git clone git@github.com:flexwang2/lfstest.git` to a new directory. From the command output, it looks like I am downloading `320.10 MiB | 60.33 MiB/s`, which means I am downloading the raw files, not pointer files. Anything wrong here? Here is the repository: https://github.com/flexwang2/lfstest
<img width="298" alt="image" src="https://user-images.githubusercontent.com/102630391/193080783-38bb92b4-47bc-461c-93c7-e681162e714f.png">

**Reply by bk2204** [reply]
Hey,

The “Filtering content” message means Git LFS is being invoked as part of the checkout process and the LFS files are being downloaded and smudged into the tree.  If you ran the clone with the environment variable `GIT_LFS_SKIP_SMUDGE=1`, then no LFS files would be downloaded, and you'd get only pointer files.

**Reply by flexwang2** [reply]
> Hey,
> 
> The “Filtering content” message means Git LFS is being invoked as part of the checkout process and the LFS files are being downloaded and smudged into the tree. If you ran the clone with the environment variable `GIT_LFS_SKIP_SMUDGE=1`, then no LFS files would be downloaded, and you'd get only pointer files.

Ohhhh, thanks!
Dumb q, if I have `GIT_LFS_SKIP_SMUDGE=1`, what would happen if I `cat large.bin` when there is only pointer files, then git would automatically download the large one or I have to run some command to pull the large file myself?

**Reply by bk2204** [reply]
If you have only pointer files, then `cat large.bin` will show only the pointer file.  If you want the large file, you'd run `git lfs pull` to download those objects and check them out in the working tree.

**Reply by flexwang2** [reply]
ty @bk2204 it helps a lot!
